File: postgraduate/src/main/java/com/postgraduate/service/TeacherService.java
package com.postgraduate.service;

import java.util.List;

import com.postgraduate.pojo.TbTeacher;

import entity.PageResult;

public interface TeacherService {
	/**
	 * 查询全部导师
	 * @return
	 */
	public List<TbTeacher> selectAll();
	/**
	 * 查询单个导师
	 * @return
	 */
	public TbTeacher selectOne(int id);
	/**
	 * 模糊查询导师
	 * @return
	 */
	public List<TbTeacher> selectPart(TbTeacher tbTeacher);
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	/**
	 * 模糊查询返回分页列表
	 * @return
	 */
	public PageResult findPage(TbTeacher tbTeacher,int pageNum,int pageSize);
	/**
	 * 新增加导师
	 * @return
	 */
	public void insert(TbTeacher tbTeacher);
	/**
	 * 修改导师信息
	 * @return
	 */
	public void update(TbTeacher tbTeacher);
	/**
	 * 修改导师密码
	 * @return
	 */
	public void updatePassById(TbTeacher tbTeacher);
}

File: postgraduate/src/main/webapp/js/service/TeacherService.js
app.service("TeacherService",function($http){
	//查询全部导师
	this.selectAll=function(){
		return $http.get('../Teacher/selectAll.do');
	}
	//单条查询
	this.selectOne=function(id){
		return $http.get('../Teacher/selectOne.do?id='+id);
	}
	//查询导师密码
	this.selectPassById=function(id){
		return $http.get('../Teacher/selectPassById.do?id='+id);
	}
	//分页
	this.findPage=function(page,rows){
		return $http.get('../Teacher/findPage.do?page='+page+'&rows='+rows);
	}
	//修改
	this.update=function(entity){
		return $http.post('../Teacher/update.do',entity);
	}
	//修改密码
	this.updatePassById=function(entity){
		return $http.post('../Teacher/updatePassById.do',entity);
	}
	//增加
	this.insert=function(entity){
		return $http.post('../Teacher/insert.do',entity);
	}
	//模糊查询分页
	this.search=function(page,rows,searchEntity){
		return $http.post('../Teacher/search.do?page='+page+'&rows='+rows,searchEntity);
	}
	this.insertLesTea=function(lesid,teaid){
		return $http.post('../LesTea/insertLesTea.do?lesid='+lesid+'&teaid='+teaid);
	}
});